dg/tests.md

```diff
@@ -0,0 +1,166 @@
+<frontmatter>
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: dg-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Introduction to Tests in CATcher
+
+1. [Jasmine Style Guide](#jasmine-style-guide)
+    1. [Test Suite](#test-suite)
+    2. [Spy](#spy)
+    3. [Resources for Jasmine](#resources-for-jasmine)
+2. [Angular TestBed Utility](#angular-testbed-utility)
+    1. [Resources for Angular TestBed](#resources-for-angular-testbed)
+3. [E2E Testing](#e2e-testing)
+    1. [Running E2E Tests](#running-e2e-tests)
+    2. [Troubleshooting conflicts between the versions of the browser and browser driver](#troubleshooting-conflicts-between-the-versions-of-the-browser-and-browser-driver)
+    3. [Protractor Configuration](#protractor-configuration)
+    4. [How the E2E tests work](#how-the-e2e-tests-work)
+
+## Jasmine Style Guide
+
+Jasmine is a behavior-driven development framework specific for testing JavaScript code. We follow the Jasmine Style Guide loosely for our tests.
+
+### Test Suite
+
+Jasmine testing involves the test suite, which is started by calling the global Jasmine function `describe`. The test suite may consist of several tests (specs) within itself, which is done by calling `it`. One main guideline is that a `describe` block should be created for each method / scenario under test, and an `it` block should be created for each property being verified.
```

Comment:
We can keep the following sentence (as it's a style guideline):
`One main guideline is that a describe block should be created for each method / scenario under test, and an `it` block should be created for each property being verified.`

Let's move the sentence to the paragraph where the style guide is introduced:

`
Jasmine is a behavior-driven development framework specific for testing JavaScript code. We follow the Jasmine Style Guide loosely for our tests. One main guideline is that a describe block should be created for each method / scenario under test, and an `it` block should be created for each property being verified.`
`